lte/gateway/python/integ_tests/s1aptests/test_ipv4v6_non_nat_ded_bearer_ul_tcp.py

```diff
@@ -0,0 +1,186 @@
+        )
+        router_adv = response.cast(s1ap_types.ueRouterAdv_t)
+        print(
+            "********** Received Router Advertisement for APN-%s"
+            " bearer id-%d" % (apn, router_adv.bearerId),
+        )
+        ipv6_addr = "".join([chr(i) for i in router_adv.ipv6Addr]).rstrip(
+            "\x00",
+        )
+        print("********** UE IPv6 address: ", ipv6_addr)
+        default_ipv6 = ipaddress.ip_address(ipv6_addr)
+        self._s1ap_wrapper.s1_util.update_ipv6_address(ue_id, ipv6_addr)
+
+        print(
+            "********************** Sending RAR for IMSI",
+            "".join([str(i) for i in req.imsi]),
+        )
+
+        # QoS
+        qos = {
+            "qci": 1,  # qci value [1 to 9]
+            "priority": 1,  # Range [0-255]
+            "max_req_bw_ul": 10000000,  # MAX bw Uplink
+            "max_req_bw_dl": 15000000,  # MAX bw Downlink
+            "gbr_ul": 1000000,  # GBR Uplink
+            "gbr_dl": 2000000,  # GBR Downlink
+            "arp_prio": 1,  # ARP priority
+            "pre_cap": 1,  # pre-emption capability
+            "pre_vul": 1,  # pre-emption vulnerability
+        }
+
+        flow_list = self._spgw_util.create_default_ipv6_flows()
+        policy_id = "magma"
+
+        self._sessionManager_util.send_ReAuthRequest(
+            "IMSI" + "".join([str(i) for i in req.imsi]),
+            policy_id,
+            flow_list,
+            qos,
+        )
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_ACT_DED_BER_REQ.value,
+        )
+        act_ded_ber_req_ims_apn = response.cast(
+            s1ap_types.UeActDedBearCtxtReq_t,
+        )
+        self._s1ap_wrapper.sendActDedicatedBearerAccept(
+            req.ue_id, act_ded_ber_req_ims_apn.bearerId,
+        )
+        print(
+            "************* Added dedicated bearer",
+            act_ded_ber_req_ims_apn.bearerId,
+        )
+
+        # Sleep before verifying flows
+        print("********** Sleeping for 5 seconds")
+        time.sleep(5)
+        num_ul_flows = 2
+        dl_flow_rules = {
+            default_ipv4: [],
+            default_ipv6: [flow_list],
+        }
+        # Verify if flow rules are created
+        self._s1ap_wrapper.s1_util.verify_flow_rules(
+            num_ul_flows, dl_flow_rules, ipv6_non_nat=True,
+        )
+
+        print(
+            "************************* Running UE uplink (TCP) for UE id ",
+            req.ue_id,
+        )
+        with self._s1ap_wrapper.configUplinkTest(req, duration=1) as test:
+            test.verify()
+        print(
+            "************************* Running UE detach for UE id ",
+            req.ue_id,
+        )
+        # Now detach the UE
+        self._s1ap_wrapper.s1_util.detach(
+            req.ue_id,
+            s1ap_types.ueDetachType_t.UE_NORMAL_DETACH.value,
+            wait_for_s1,
+        )
+
```

Comment:
We probably need some sleep after detach and before flow rule deletion verification (same for other testcases)